### ISSUE TITLE
Pollers kept alive after being disposed of

### ISSUE THREAD

**Issue by thmaterna** [zeromq/netmq#131]
Hi,

I migrated a project from clrzmq to NetMQ and needed a way to Poll just once at a time. I understand that NetMQ is not meant to work that way but it makes it easier to migrate the library before refactoring into a more NetMQ-like logic.

I first tried to start/stop, then restart the Poller, but that doesn't work since the `m_cancellationTokenSource` in the Poller is readonly. There is no way, once the cancellation has been requested in `Poller.Stop()`, to reset it. The next call to `Start` will not work as intended since the while condition `while (!m_cancellationTokenSource.IsCancellationRequested)` will always be false.  I suppose this could be a separate issue by itself, unless a Poller is explicitly meant to be started only once (?).

I managed to get what I wanted that by writing an extension method like this:

``` C#
public static void Poll(this NetMQSocket[] sockets, int timeoutMs)
{
    using (var poller = new Poller(sockets))
    {
        poller.PollTimeout = timeoutMs;
        var pollerTask = Task.Factory.StartNew(poller.Start);

        // avoid race condition
        while(!poller.IsStarted)
            Thread.Sleep(2);

        poller.Stop(true);
    }
}
```

This caused a managed memory leak: the poller registers an event handler in AddSocket: `socket.EventsChanged += OnSocketEventsChanged;`, but it does NOT unregister the handler when it is disposed of. So each time this method is called, a new Poller is created and kept alive forever by the sockets.

Fortunately, `OnSocketEventsChanged` is very fast and at least that doesn't add a performance problem to the memory problem (since now each time the `socket.EventsChanged` is fired, all the past Poller are called back). My workarounf for now is to add the following code inside my `using` block in the above method.

``` C#
        foreach (var t in sockets)
        {
            poller.RemoveSocket(t);
        }
```

A much better way would be to have the `Poller.Dispose` unregister cleanly:

``` C#
foreach (var t in m_sockets.ToList())// make a copy of the list because RemoveSocket modifies m_sockets
{
    RemoveSocket(t);
}
```
